### PR TITLE
wesher: init at 0.2.6

### PR DESCRIPTION
**Pull request by tylerjl**
###### Description of changes

[Wesher](https://github.com/costela/wesher) is a very simple wireguard mesh daemon. Think of it as a local-only, clustered, and zero-config Tailscale. I've been running wesher in my homelab using this package for a few months so I think it's good to go into the main package set.

I also tossed in my name into `maintainers-list.nix`. Hopefully I can continue to contribute and help out!

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- ~[ ] Tested, as applicable:~
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- ~[ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)~
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- ~[22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)~
  - ~[ ] (Package updates) Added a release notes entry if the change is major or breaking~
  - ~[ ] (Module updates) Added a release notes entry if the change is significant~
  - ~[ ] (Module addition) Added a release notes entry if adding a new NixOS module~
  - ~[ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes~
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/wesher/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{ lib, buildGoModule, fetchFromGitHub }:
```
Comment (IvarWithoutBones) [inline pkgs/servers/wesher/default.nix:1]: I would prefer for these to be separated by newlines, that makes future diffs a lot easier to read.
```suggestion
{ lib
, buildGoModule
, fetchFromGitHub
}:
```

**Review by IvarWithoutBones — COMMENTED** [review]
Thanks for the PR! Left you a few (mostly style related 😅) suggestions

File: pkgs/servers/wesher/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  vendorSha256 = "sha256-BZzhBC4C0OoAxUEDROkggCQF35C9Z4+0/Jk0ZD8Hz1s=";
+
+  ldflags = [
+    "-X main.version=${version}"
+  ];
+
+  meta = with lib; {
+    description = "Wireguard overlay mesh network manager";
+    homepage = "https://github.com/costela/wesher";
+    license = licenses.gpl3;
```
[the remark this answers is not in the document]
Comment (tylerjl) [reply pkgs/servers/wesher/default.nix]: Noted, thank you!

File: pkgs/servers/wesher/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "wesher";
+  version = "0.2.6";
+
+  src = fetchFromGitHub {
+    owner = "costela";
+    repo = pname;
```
[the remark this answers is not in the document]
Comment (tylerjl) [reply pkgs/servers/wesher/default.nix]: Done!

File: pkgs/servers/wesher/default.nix (hunk 1 above)
Comment (tylerjl) [reply pkgs/servers/wesher/default.nix]: I definitely agree with this, changes in https://github.com/NixOS/nixpkgs/pull/194169/commits/2f737c64af82f1e0983e8a619711fe1c8e28d8cf

**Comment by tylerjl** [comment]
Thank you for the review @IvarWithoutBones :pray: Stylistic changes committed in https://github.com/NixOS/nixpkgs/pull/194169/commits/2f737c64af82f1e0983e8a619711fe1c8e28d8cf

File: pkgs/servers/wesher/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    owner = "costela";
+    repo = "wesher";
+    rev = "v${version}";
+    sha256 = "sha256-EIajvcBhS5G9dJzRgXhnD1QKOAhmzngdyCU4L7itT8U=";
+  };
+
+  vendorSha256 = "sha256-BZzhBC4C0OoAxUEDROkggCQF35C9Z4+0/Jk0ZD8Hz1s=";
+
+  ldflags = [
+    "-X main.version=${version}"
```
Comment (SuperSandro2000) [inline pkgs/servers/wesher/default.nix:20]:
```suggestion
    "-s" "-w" "-X main.version=${version}"
```

**Review by SuperSandro2000 — COMMENTED** [review]
Please squash the review commit into the corresponding init commit. Then we can merge this.

**Comment by tylerjl** [comment]
@SuperSandro2000 thanks! Squashed the review commits and rebased onto latest `master`.

**Comment by SuperSandro2000** [comment]
Can you bump ``golang.org/x/sys/unix`` to try to resolve the darwin build failure?

File: pkgs/servers/wesher/default.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+  ldflags = [
+    "-s" "-w" "-X main.version=${version}"
+  ];
+
+  meta = with lib; {
+    description = "Wireguard overlay mesh network manager";
+    homepage = "https://github.com/costela/wesher";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ tylerjl ];
```
Comment (tylerjl) [inline pkgs/servers/wesher/default.nix:27]:
```suggestion
    maintainers = with maintainers; [ tylerjl ];
    platforms   = platforms.linux;
```

Comment (tylerjl) [reply pkgs/servers/wesher/default.nix:27]: @SuperSandro2000 so I tried building against `GOOS=darwin` on varying go versions (1.17, 1.18, 1.19) on different revisions of the upstream code but it looks like there isn't a combination that compiles successfully for darwin - coupled with the fact that the [releases page](https://github.com/costela/wesher/releases/tag/v0.2.6) only lists Linux artifacts, I think scoping `wesher` down solely to `platforms.linux` is probably the right choice. Would this change be acceptible?

Comment (balsoft) [reply pkgs/servers/wesher/default.nix:27]: @tylerjl please do mark it for `platforms.linux`

Comment (SuperSandro2000) [reply pkgs/servers/wesher/default.nix:27]: Yeah, we can do that but I am pretty sure that bumping x/sys fixes the current error. Did you try that and there where others?

Comment (tylerjl) [reply pkgs/servers/wesher/default.nix:27]: @balsoft thanks, it seems like the right approach to me :+1: 

@SuperSandro2000 I bumped the dependencies but ended up chasing more errors - I may head to the upstream repo and ask for a new release to get newer deps/updated changes which may help here.

**Pull request merged**